cogs/base.py

```diff
@@ -1,57 +1,50 @@
-# from functools import wraps
-from discord import app_commands
+import discord
+
 from discord.ext import commands
 
-from components.bot import RecruitBot
-from components.config.config_manager import configInstance
+from components.bot import Bot
 
-# def guilds_wrapper(f):
-#     @wraps(f)
-#     def _impl(self, *method_args, **method_kwargs):
-#         app_commands.guilds(configInstance.data.guild)
-#         return f(*method_args, **method_kwargs)
-#     return _impl
 
-class Base(commands.Cog):
-    def __init__(self, bot: RecruitBot):
-        self.bot = bot
+def is_authorized():
+    def predicate(ctx: commands.Context):
+        if not (discord.utils.get(ctx.author.roles, name="Admin") or discord.utils.get(ctx.author.roles, name="Recruit Manager")):
```

Comment:
For admin, we should be checking for the Administrator permission, rather than the 'Admin' role.